src/librustc/ty/mod.rs

```diff
@@ -592,7 +591,8 @@ impl<'tcx> serialize::UseSpecializedDecodable for &'tcx Slice<Ty<'tcx>> {}
 impl<T> Slice<T> {
     pub fn empty<'a>() -> &'a Slice<T> {
         unsafe {
-            mem::transmute(slice::from_raw_parts(0x1 as *const T, 0))
+            let slice = slice::from_raw_parts(0x1 as *const T, 0);
+            &*(slice as *const _ as *const _)
```

Comment:
This one in particular I don't know why transmute isn't the best way. We need some way to do the `&[X] -> &NewType([X])` cast.

Reply:
This still performs the same cast, just without transmuting the fat pointer. Can you explain further why transmute would be preferred here?